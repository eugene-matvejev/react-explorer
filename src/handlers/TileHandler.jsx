import React from 'react';
import Tile from '../components/Tile';
import HealthCheck from '../components/HealthCheck';

const fetchStatuses = callBack => callBack([
    {
      id: 1,
      status : "red",
      name : "Health Check 1",
      parent: null
    },
    {
      id: 2,
      status: "green",
      name: "Health Check 2",
      parent: null
    },
    {
      id: 3,
      status: "amber",
      name: "Health Check 3",
      parent: null
    }
]);

export default class TileHandler extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      statuses: [],
      isExpanded: false,
      isExpandedIndex: 0,
    }

    this.onClick = this.onClick.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }
  
  
  componentDidMount() {
    fetchStatuses(statusArray => 
      this.setState({
        statuses: statusArray,
    }));
  }

  setExpandedState(tileIndex) {
    const { isExpanded } = this.state;
    this.setState({
      isExpanded: !isExpanded,
      isExpandedIndex: tileIndex
    });
  }

  onClick(tileId) {
    this.setExpandedState(tileId);
  }

  handleClose(){
    this.setExpandedState(0);
  }

  render() {
    const {statuses, isExpanded, isExpandedIndex} = this.state;
    const dialogStatus = statuses.filter(status => status.id === isExpandedIndex);
    let className;
    return (
      <section className={`tile-handler ${className}`}>
        {statuses.map(
          (status) => <Tile key = {status.id} {...status} onClick={this.onClick} />
        )}
        {this.state.isExpanded ? 
          <HealthCheck { ...{ isExpanded, ...dialogStatus[0] }} handleClose={this.handleClose}/>
          : null
        }
      </section>
    );
  }
}