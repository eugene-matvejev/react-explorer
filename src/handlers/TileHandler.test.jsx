import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TileHandler from './TileHandler';

configure({ adapter: new Adapter() });

describe('TileHandler', () => {

  let mountedTileHandler;

  beforeEach(() => {
    mountedTileHandler = shallow(<TileHandler />);
  });

  describe('renders', () => { 
    it('has no props', () => {
      expect(mountedTileHandler).toMatchSnapshot();
    });
  });

  describe('lifecycle method', () => { 
    it.skip('mounts component with placeholder data', () => {
      
    });
  });

});