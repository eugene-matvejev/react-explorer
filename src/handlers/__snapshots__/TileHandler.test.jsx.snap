// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileHandler renders has no props 1`] = `
<section
  className="tile-handler undefined"
>
  <Tile
    className=""
    id={1}
    key="1"
    name="Health Check 1"
    onClick={[Function]}
    parent={null}
    status="red"
  />
  <Tile
    className=""
    id={2}
    key="2"
    name="Health Check 2"
    onClick={[Function]}
    parent={null}
    status="green"
  />
  <Tile
    className=""
    id={3}
    key="3"
    name="Health Check 3"
    onClick={[Function]}
    parent={null}
    status="amber"
  />
</section>
`;
